bentoml/deployment/aws_lambda/__init__.py

```diff
@@ -0,0 +1,425 @@
+                        'BENTOML_ARTIFACTS_PREFIX': artifacts_prefix,
+                        'BENTOML_API_NAME': api_name,
+                    }
+                },
+            },
+        }
+
+    yaml.dump(sam_config, Path(template_file_path))
+
+    # We add Outputs section separately, because the value should not
+    # have "'" around !Sub
+    with open(template_file_path, 'a') as f:
+        f.write(
+            """\
+  EndpointUrl:
+    Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod"
+    Description: URL for endpoint
+"""
+        )
+    return template_file_path
+
+
+def _cleanup_s3_bucket(bucket_name, region):
+    s3_client = boto3.client('s3', region)
+    s3 = boto3.resource('s3')
+    try:
+        logger.debug('Removing all objects inside bucket %s', bucket_name)
+        s3.Bucket(bucket_name).objects.all().delete()
+        logger.debug('Deleting bucket %s', bucket_name)
+        s3_client.delete_bucket(Bucket=bucket_name)
+    except ClientError as e:
+        if e.response and e.response['Error']['Code'] == 'NoSuchBucket':
+            # If there is no bucket, we just let it silently fail, dont have to do
+            # any thing
+            return
+        else:
+            raise e
+
+
+class AwsLambdaDeploymentOperator(DeploymentOperatorBase):
+    def apply(self, deployment_pb, yatai_service, prev_deployment):
+        try:
+            ensure_sam_available_or_raise()
+            ensure_docker_available_or_raise()
+            deployment_spec = deployment_pb.spec
+
+            bento_pb = yatai_service.GetBento(
+                GetBentoRequest(
+                    bento_name=deployment_spec.bento_name,
+                    bento_version=deployment_spec.bento_version,
+                )
+            )
+            if bento_pb.bento.uri.type not in (BentoUri.LOCAL, BentoUri.S3):
+                raise BentoMLException(
+                    'BentoML currently not support {} repository'.format(
+                        bento_pb.bento.uri.type
+                    )
+                )
+
+            return self._apply(
+                deployment_pb,
+                bento_pb,
+                yatai_service,
+                bento_pb.bento.uri.uri,
+                prev_deployment,
+            )
+        except BentoMLException as error:
+            return ApplyDeploymentResponse(status=exception_to_return_status(error))
+
+    def _apply(
+        self, deployment_pb, bento_pb, yatai_service, bento_path, prev_deployment
+    ):
+        if loader._is_remote_path(bento_path):
+            with loader._resolve_remote_bundle_path(bento_path) as local_path:
+                return self._apply(
+                    deployment_pb, bento_pb, yatai_service, local_path, prev_deployment
+                )
+
+        try:
+            deployment_spec = deployment_pb.spec
+            aws_config = deployment_spec.aws_lambda_operator_config
```

Comment:
rename to deployment_config  or aws_lambda_config? aws_config is a bit confusing